src/compilers/bf/sanity-check.js

```diff
@@ -1,18 +1,15 @@
 const fs = require('../../utils').fs;
 
-module.exports = (argv) =>
-  new Promise((resolve) => {
-    if (!argv.i || !argv.o) {
-      console.log('Usage: node src/compilers/bf -i {infile} -o {outfile}');
-      process.exit(1);
-    }
+module.exports = (argv) => {
+  if (!argv.i || !argv.o) {
+    console.log('Usage: node src/compilers/bf -i {infile} -o {outfile}');
+    process.exit(1);
+  }
 
-    fs.statAsync(argv.i)
-      .then(stats => {
-        if (stats.isFile()) {
-          resolve();
-        } else {
-          throw new Error(`Brainfuck file ${argv.i} is not valid. Exiting...`);
-        }
-      });
-  });
+  return fs.statAsync(argv.i)
+    .then(stats => {
+      if (!stats.isFile()) {
+        throw new Error(`Brainfuck file ${argv.i} is not valid. Exiting...`);
```

Comment:
While you're changing things could you add a process.exit(1) after the thrown error?